internal/store/endpoint.go

```diff
@@ -102,6 +103,9 @@ var (
 					},
 				}
 			}),
+			AggregateBy: map[string]aggregation.Aggregation{
+				"namespace": aggregation.ByLabels("namespace"),
```

Comment:
This is just a thought, that maybe some of the generality that you have here in the "ByLabels" clause, that it might be possible to make that accessible through the startup arguments for KSM. (We would not need that for internal purposes, but it might be helpful for an OSS PR.)